src/webpack/common/internal.tsx

```diff
@@ -16,19 +16,19 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
-import { LazyComponent } from "@utils/react";
+import { LazyComponent, LazyComponentWrapper } from "@utils/react";
 
 // eslint-disable-next-line path-alias/no-relative
 import { FilterFn, filters, lazyWebpackSearchHistory, waitFor } from "../webpack";
 
-export function waitForComponent<T extends React.ComponentType<any> = React.ComponentType<any> & Record<string, any>>(name: string, filter: FilterFn | string | string[]): T {
+export function waitForComponent<T extends React.ComponentType<any> = React.ComponentType<any> & Record<string, any>>(name: string, filter: FilterFn | string | string[]) {
     if (IS_REPORTER) lazyWebpackSearchHistory.push(["waitForComponent", Array.isArray(filter) ? filter : [filter]]);
 
     let myValue: T = function () {
         throw new Error(`Vencord could not find the ${name} Component`);
     } as any;
 
-    const lazyComponent = LazyComponent(() => myValue) as T;
+    const lazyComponent = LazyComponent(() => myValue) as LazyComponentWrapper<T>;
```

Comment:
```suggestion
    const lazyComponent = LazyComponent(() => myValue);
```

I think this isn't needed anymore

Reply:
typescript is goofball